src/state/assets/userAssets.ts

```diff
@@ -180,21 +180,26 @@ export const userAssetsStore = createRainbowStore<UserAssetsState>(
       }
     },
 
-    getHighestValueAsset: (usePreferredNetwork = true) => {
-      const preferredNetwork = usePreferredNetwork ? swapsStore.getState().preferredNetwork : undefined;
+    getHighestValueEth: () => {
+      const preferredNetwork = swapsStore.getState().preferredNetwork;
       const assets = get().userAssets;
 
-      if (preferredNetwork && get().getChainsWithBalance().includes(preferredNetwork)) {
-        // Find the highest-value asset on the preferred network
-        for (const [, asset] of assets) {
-          if (asset.chainId === preferredNetwork) {
+      let highestValueEth: ParsedSearchAsset | null = null;
+
+      for (const [, asset] of assets) {
+        if (asset.mainnetAddress === ETH_ADDRESS) {
+          // if there is a preferredNetwork & this eth asset is on that network, return it
+          if (preferredNetwork && asset.chainId === preferredNetwork) {
             return asset;
+            // if this eth asset is not on the preferredNetwork, update highestValueEth if applicable
+          } else if (!highestValueEth || asset.balance > highestValueEth.balance) {
+            highestValueEth = asset;
           }
         }
       }
 
-      // If no preferred network asset, return the highest-value asset
-      return assets.values().next().value || null;
+      // If no eth on the preferred network, return the highest value eth asset
+      return highestValueEth;
     },
```

Comment:
I think there are ways to improve the readability of this. Mainly filtering before looping over all assets. Something like this:
```ts
getHighestValueEth = () => {
  const preferredNetwork = swapsStore.getState().preferredNetwork;
  const assets = Array.from(get().userAssets.values());
  
  const ethAssets = assets.filter(asset => asset.mainnetAddress === ETH_ADDRESS);

 // if no eth assets present, return null? Maybe there's a better fallback option here
  if (!ethAssets.length) {
    return null;
  }

  if (preferredNetwork) {
    const preferredEthAsset = ethAssets.find(asset => asset.chainId === preferredNetwork);
    if (preferredEthAsset) {
      return preferredEthAsset;
    }
  }

  return ethAssets.reduce((highest, current) => 
    current.balance > highest.balance ? current : highest
  );
};
```

Reply:
very nice

Reply:
might be negligible for wallets with few assets but that's adding a lot of unnecessary steps imo

vs doing it in a single pass and working with the existing data structure:

```typescript
getHighestValueEth: () => {
  const preferredNetwork = swapsStore.getState().preferredNetwork;
  let highestValueEth = null;

  for (const [, asset] of assets) {
    if (asset.mainnetAddress !== ETH_ADDRESS) continue;

    if (preferredNetwork && asset.chainId === preferredNetwork) {
      return asset;
    }

    if (!highestValueEth || asset.balance > highestValueEth.balance) {
      highestValueEth = asset;
    }
  }

  return highestValueEth;
}
```